packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -320,6 +350,11 @@ class Plugin {
     if (isInvalid(WindowsPlugin.kConfigKey, WindowsPlugin.validate)) {
       errors.add('Invalid "windows" plugin specification.');
     }
+    yaml.forEach((dynamic key, dynamic value) {
+      if (key is String && key.startsWith('x-')) {
+        isInvalid(key, CustomEmbedderPlugin.validate);
```

Comment:
I'm not sure we can meaningfully validate custom plugin sections; they could contain anything.